src/test/kotlin/test/steps/IOSteps.kt

```
@@ -23,7 +24,8 @@ class IOSteps {
 
     @And("the file {string} named {string} with content")
     fun createFileWithContent(variableName: String, fileName: String, content: String) {
-        val file = tempFile.toPath().resolve(fileName).createFile().apply { writeText(content) }
+        val file = tempFile.toPath().resolve(fileName).apply { deleteIfExists() }
```

lets remove this create a file should fail if a file already exits

---

is not clear to me why should i remove this part. It is used in case we want to overwrite a file

---

a test failing because it is trying to create a file that already exists is the expected scenario,  that means this is is not seting the conditions for its execution or another test is not cleaning its data. We should not create steps ignores this errors but rather fix errors in the root cause.

---

agree, thanks for this comment